src/clj/puppetlabs/services/jruby/jruby_puppet_service.clj

```diff
@@ -42,6 +42,13 @@
             (assoc :pool-context pool-context)
             (assoc :borrow-timeout (:borrow-timeout config))
             (assoc :event-callbacks (atom []))))))
+  (stop
+   [this context]
+   (let [{:keys [pool-context]} (tk-services/service-context this)
+         flush-complete? (promise)]
+     (jruby-agents/send-flush-pool-for-shutdown! pool-context flush-complete?)
+     @flush-complete?)
```

Comment:
> do you have any idea what sort of timeout would make sense here?

Not really.  I'd be tempted to go with something like 1 minute.  I wonder if @camlow325 or @cprice404 have an opinion on this.